PlayCover/ViewModel/StoreVM.swift

```diff
@@ -29,14 +36,26 @@ class StoreVM: ObservableObject {
             encode()
         }
     }
+    @Published var keymappingSources: [SourceData] {
+        didSet {
+            encode(source: keymappingSources)
+        }
+    }
 
     let sourcesUrl: URL
+    let keymappingSourcesUrl: URL
 
     @discardableResult
-    public func decode() -> Bool {
+    public func decode(source: [SourceData]? = nil) -> Bool {
```

Comment:
I'm not sure about this approach. I think it might be better to just give them each separate functions.